src/Core/src/Response.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core;
+
+use AsyncAws\Core\Exception\Http\ClientException;
+use AsyncAws\Core\Exception\Http\HttpException;
+use AsyncAws\Core\Exception\Http\NetworkException;
+use AsyncAws\Core\Exception\Http\RedirectionException;
+use AsyncAws\Core\Exception\Http\ServerException;
+use AsyncAws\Core\Exception\RuntimeException;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * The response provides a facade to manipulate HttpResponses.
+ */
+class Response
```

Comment:
We could add the `@final` annotation on this response and the request. 

That will show that one should not extend these classes, but one can still mock them properly.

Reply:
good idea. thanks